src/main/kotlin/no/nav/integrasjon/api/v1/Streams.kt

```diff
@@ -0,0 +1,72 @@
+package no.nav.integrasjon.api.v1
+
+import io.ktor.application.call
+import io.ktor.http.HttpStatusCode
+import io.ktor.locations.Location
+import io.ktor.response.respond
+import io.ktor.routing.Routing
+import no.nav.integrasjon.FasitProperties
+import no.nav.integrasjon.api.nielsfalk.ktor.swagger.BasicAuthSecurity
+import no.nav.integrasjon.api.nielsfalk.ktor.swagger.Group
+import no.nav.integrasjon.api.nielsfalk.ktor.swagger.badRequest
+import no.nav.integrasjon.api.nielsfalk.ktor.swagger.ok
+import no.nav.integrasjon.api.nielsfalk.ktor.swagger.post
+import no.nav.integrasjon.api.nielsfalk.ktor.swagger.securityAndReponds
+import no.nav.integrasjon.api.nielsfalk.ktor.swagger.serviceUnavailable
+import no.nav.integrasjon.api.nielsfalk.ktor.swagger.unAuthorized
+import org.apache.kafka.clients.admin.AdminClient
+import org.apache.kafka.common.acl.AccessControlEntry
+import org.apache.kafka.common.acl.AclBinding
+import org.apache.kafka.common.acl.AclOperation
+import org.apache.kafka.common.acl.AclPermissionType
+import org.apache.kafka.common.resource.PatternType
+import org.apache.kafka.common.resource.ResourcePattern
+import org.apache.kafka.common.resource.ResourceType
+import java.util.concurrent.TimeUnit
+
+private const val swGroup = "Streams"
+
+enum class PostStreamStatus {
+    ERROR,
+    OK
+}
+
+@Group(swGroup)
+@Location("$STREAMS/")
+class PostStream
+data class PostStreamBody(val applicationName: String, val user: String)
+
+data class PostStreamResponse(val status: PostStreamStatus, val message: String)
+
+fun Routing.streamsAPI(adminClient: AdminClient?, fasitConfig: FasitProperties) {
```

Comment:
If we do it this way the ACLs specified in topics/groups/oneshot should also be applied to the sub topics for this kafka stream? Since we're doing a different ResourcePattern then the one before it, would it be possible that the old one wont be overwritten? If so we need to handle that too

Reply:
https://github.com/navikt/kafka-adminrest/blob/35e4c217b4c101756f256e345ade29790cbe4875/src/main/kotlin/no/nav/integrasjon/ldap/LDAPGroup.kt#L383

Reply:
The oneshot-api is focused on _topics_ where-as a kafka consumer can create many different internal topics. Because groups are created per topic this would be harder to implement, I think.

If my app joins two topics together, the consumer will automatically create a few topics to accommodate this.